services/cucumber/test/TestCucumberWireProtocolController.cpp

```diff
@@ -0,0 +1,72 @@
+#include "infra/timer/test_helper/ClockFixture.hpp"
+#include "infra/util/BoundedString.hpp"
+#include "services/cucumber/CucumberContext.hpp"
+#include "services/cucumber/CucumberRequestHandlers.hpp"
+#include "services/cucumber/CucumberWireProtocolController.hpp"
+#include "services/cucumber/CucumberWireProtocolParser.hpp"
+#include "services/network/test_doubles/ConnectionMock.hpp"
+#include "gmock/gmock.h"
+#include <algorithm>
+
+namespace
+{
+    class CucumberScenarioRequestHandlerMock
+        : public services::CucumberScenarioRequestHandler
+    {
+    public:
+        MOCK_METHOD(void, BeginScenario, (infra::JsonArray & tags, const infra::Function<void()>& onDone), (override));
+        MOCK_METHOD(void, EndScenario, (const infra::Function<void()>& onDone), (override));
+    };
+
+    class CucumberWireProtocolControllerTest
+        : public testing::Test
+        , public infra::ClockFixture
```

Comment:
I don't expect ClockFixture is necessary. Is an EventDispatcher fixture sufficient/necessary?

Reply:
The controller has a dependency on `services::CucumberContext`, which has a `infra::TimerSingleShot`